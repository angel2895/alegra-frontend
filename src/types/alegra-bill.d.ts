export interface AlegraBill {
  date: Date;
  dueDate: Date;
  client: number;
  seller: number;
  items: Item[];
}

export interface Item {
  id: number;
  description: string;
  price: number;
  quantity: number;
}
