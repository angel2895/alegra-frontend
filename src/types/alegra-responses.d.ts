export interface AlegraResponseProduct {
  id: string;
  category: Category;
  name: string;
  description: string;
  reference: string;
  status: string;
  price: Price[];
  inventory: Inventory;
  tax: any[];
  customFields: any[];
  productKey: null;
  type: string;
  itemType: null;
  settingsOnPos: SettingsOnPos;
}

export interface Category {
  id: string;
  name: string;
}

export interface Inventory {
  unit: string;
}

export interface Price {
  idPriceList: number;
  name: string;
  type: string;
  price: string;
  currency: Currency;
}

export interface Currency {
  code: string;
  symbol: string;
}

export interface SettingsOnPos {
  hide: null;
}

export interface AlegraResponseBill {
  id: string;
  date: Date;
  dueDate: Date;
  datetime: Date;
  observations: null;
  anotation: null;
  termsConditions: string;
  status: string;
  client: Client;
  numberTemplate: NumberTemplate;
  subtotal: number;
  discount: number;
  tax: number;
  total: number;
  totalPaid: number;
  balance: number;
  decimalPrecision: string;
  warehouse: Warehouse;
  operationType: string;
  seller: Seller;
  priceList: PriceList;
  items: Item[];
  costCenter: null;
}

export interface Client {
  id: string;
  name: string;
  identification: string;
  phonePrimary: null;
  phoneSecondary: null;
  fax: null;
  mobile: string;
  email: string;
  address: Address;
}

export interface Address {
  state: string;
  city: string;
  district: string;
  urbanization: null;
  ubigeoCode: string;
  address: string;
}

export interface Item {
  name: string;
  description: null;
  price: number;
  discount: number;
  reference: null;
  quantity: number;
  id: number;
  productKey: null;
  unit: Unit;
  tax: any[];
  total: number;
}

export enum Unit {
  Unit = "unit",
}

export interface NumberTemplate {
  id: string;
  prefix: string;
  number: string;
  text: null;
  documentType: string;
  fullNumber: string;
  formattedNumber: string;
}

export interface PriceList {
  id: number;
  name: string;
}

export interface Seller {
  id: number;
  name: string;
  identification: string;
  observations: string;
}

export interface Warehouse {
  id: string;
  name: string;
}
