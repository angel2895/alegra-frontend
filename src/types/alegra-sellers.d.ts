export interface AlegraSeller {
  id: number;
  name: string;
  identification: string;
  observations: string;
  status: Status;
}

export enum Status {
  Active = "active",
}
