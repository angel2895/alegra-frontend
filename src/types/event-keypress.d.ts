export interface EventKeyPress {
  isTrusted: boolean;
  altKey: boolean;
  bubbles: boolean;
  cancelBubble: boolean;
  cancelable: boolean;
  charCode: Number;
  code: String;
}
