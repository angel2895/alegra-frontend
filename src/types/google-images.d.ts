export interface GoogleImages {
  search_metadata: SearchMetadata;
  search_parameters: SearchParameters;
  search_information: SearchInformation;
  suggested_searches: SuggestedSearch[];
  images_results: GoogleImage[];
}

export interface GoogleImage {
  position: number;
  thumbnail: string;
  source: string;
  title: string;
  link: string;
  original: string;
  is_product: boolean;
  in_stock?: boolean;
}

export interface SearchInformation {
  image_results_state: string;
  query_displayed: string;
}

export interface SearchMetadata {
  id: string;
  status: string;
  json_endpoint: string;
  created_at: string;
  processed_at: string;
  google_url: string;
  raw_html_file: string;
  total_time_taken: number;
}

export interface SearchParameters {
  engine: string;
  q: string;
  google_domain: string;
  ijn: string;
  device: string;
  tbm: string;
}

export interface SuggestedSearch {
  name: string;
  link: string;
  chips: string;
  serpapi_link: string;
  thumbnail: string;
}
