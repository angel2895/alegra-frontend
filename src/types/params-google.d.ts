export interface ParamsGoogle {
  api_key: string;
  tbm: string;
  ijn: number;
  q?: string;
}
