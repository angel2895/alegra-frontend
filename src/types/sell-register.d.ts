export interface SellRegister {
  id: number;
  name: string;
  points: number;
  products: Image[];
}

export interface SellRegisterShort {
  id: number;
  name: string;
  points: number;
}

export interface Image {
  key: string;
  url: string;
  title: string;
}

export interface ImageSelled {
  id: number;
  name: string;
  key: string;
  url: string;
  title: string;
}
