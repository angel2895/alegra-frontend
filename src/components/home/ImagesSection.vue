<template>
  <section class="images">
    <loader-line v-if="loadLine"></loader-line>
    <div class="images-container">
      <photo
        v-for="(seller, index) in getSellers"
        :key="index"
        :seller="seller"
        :image="getImages[index]"
      ></photo>
    </div>
  </section>
</template>
<script lang="ts">
// Libraries
import { Options, Vue } from "vue-class-component";
import { mapGetters } from "vuex";
// Types
import { GoogleImage } from "@/types/google-images";
import { AlegraSeller } from "@/types/alegra-seller";
// Components
import Photo from "@/components/home/images-section/Photo.vue";
import LoaderLine from "@/components/home/images-section/LoaderLIne.vue";

@Options({
  components: { Photo, LoaderLine },
  watch: {
    getLoading(loading: boolean): void {
      if (loading) {
        this.loadLine = true;
      } else {
        this.loadLine = false;
      }
    },
  },
  computed: mapGetters(["getImages", "getSellers", "getLoading"]),
})
export default class ImagesSection extends Vue {
  getImages!: GoogleImage[];
  getSellers!: AlegraSeller[];
  loadLine = true;
}
</script>

<style scoped>
.images-container {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-evenly;
  align-items: center;
  margin-top: 10px;

  background-color: var(--background-color);
}
</style>
