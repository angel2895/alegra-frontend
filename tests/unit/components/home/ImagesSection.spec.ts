import { shallowMount } from "@vue/test-utils";
import { createStore } from "vuex";
// Components
import ImagesSection from "@/components/home/ImagesSection.vue";
// Types
import { GoogleImage } from "@/types/google-images";
import { AlegraSeller } from "@/types/alegra-seller";

const store = createStore({
  state() {
    return {
      images: [] as GoogleImage[],
      isLoading: true,
      sellers: [] as AlegraSeller[],
    };
  },
  getters: {
    getImages: (state) => {
      const googleImage1: GoogleImage = {
        position: 1,
        thumbnail:
          "https://serpapi.com/searches/6215dbbd2c68789dfda5f456/images/05109f1d08e78809ac8ac33e43ac30be36fe799ed67b01bb8b42b1203f73ed6e.png",
        source: "mobile.twitter.com",
        title: "Apple (@Apple) | Twitter",
        link: "https://mobile.twitter.com/apple",
        original:
          "https://pbs.twimg.com/profile_images/1283958620359516160/p7zz5dxZ.jpg",
        is_product: false,
      };
      const googleImage2: GoogleImage = {
        position: 2,
        thumbnail:
          "https://serpapi.com/searches/6215dbbd2c68789dfda5f456/images/05109f1d08e7880997a7d55a45910361fe68c6ecad37992ee517563067e91c80.png",
        source: "apple.com",
        title: "iPhone - Apple (ID)",
        link: "https://www.apple.com/id/iphone/",
        original:
          "https://www.apple.com/ac/structured-data/images/knowledge_graph_logo.png?202202032151",
        is_product: false,
      };
      (state as any).images.push(googleImage1);
      (state as any).images.push(googleImage2);
      return (state as any).images;
    },
    getSellers: (state) => {
      const alegraSeller1 = {
        id: 8,
        name: "Saul Serrano",
        identification: "89ea9a867a1b071928566e4990ebe503",
        observations: "Good Seller",
        status: "active",
      };
      const alegraSeller2 = {
        id: 9,
        name: "Felisa Olmos",
        identification: "fdbb171d70cf3c80fe03d6158e2d8a0f",
        observations:
          'Felisa Olmos es un/una gran vendedor/a de nuestra empresa "Imagenes del Mundo".',
        status: "active",
      };
      (state as any).sellers.push(alegraSeller1);
      (state as any).sellers.push(alegraSeller2);
      return (state as any).sellers;
    },

    getLoading: (state) => (state as any).isLoading,
  },
});

describe("ImagesSection.vue", () => {
  it("Match with snapshot", () => {
    const wrapper = shallowMount(ImagesSection, {
      global: {
        plugins: [store],
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
  it("Photo and LoadLine are mounted in ImageSection", () => {
    const wrapper = shallowMount(ImagesSection, {
      global: {
        plugins: [store],
      },
    });
    expect(wrapper.find("photo-stub").exists()).toBeTruthy();
    expect(wrapper.find("loader-line-stub").exists()).toBeTruthy();
  });
  it("Correct properties sent to Photo", () => {
    const wrapper = shallowMount(ImagesSection, {
      global: {
        plugins: [store],
      },
    });
    const $photo = wrapper.find("photo-stub");
    expect($photo.attributes("seller")).toBe("[object Object]");
    expect($photo.attributes("image")).toBe("[object Object]");
  });
});
