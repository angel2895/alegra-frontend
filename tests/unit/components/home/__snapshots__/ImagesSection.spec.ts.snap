// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImagesSection.vue Match with snapshot 1`] = `
<section class="images">
  <loader-line-stub></loader-line-stub>
  <div class="images-container">
    <photo-stub seller="[object Object]" image="[object Object]"></photo-stub>
    <photo-stub seller="[object Object]" image="[object Object]"></photo-stub>
  </div>
</section>
`;
